validator/client/validator_propose.go

```diff
@@ -21,41 +21,42 @@ import (
 // chain node to construct the new block. The new block is then processed with
 // the state root computation, and finally signed by the validator before being
 // sent back to the beacon node for broadcasting.
-func (v *validator) ProposeBlock(ctx context.Context, slot uint64, idx string) {
+func (v *validator) ProposeBlock(ctx context.Context, slot uint64, pk string) {
 	if slot == 0 {
 		log.Info("Assigned to genesis slot, skipping proposal")
 		return
 	}
 	ctx, span := trace.StartSpan(ctx, "validator.ProposeBlock")
 	defer span.End()
-	span.AddAttributes(trace.StringAttribute("validator", fmt.Sprintf("%#x", v.keys[idx].PublicKey.Marshal())))
-	truncatedPk := idx
-	if len(idx) > 12 {
-		truncatedPk = idx[:12]
+	span.AddAttributes(trace.StringAttribute("validator", fmt.Sprintf("%#x", v.keys[pk].PublicKey.Marshal())))
+	truncatedPk := pk
+	if len(pk) > 12 {
```

Comment:
I think we have a helper that does this, we should use that here too

Reply:
Ah. Yes. Done